driver/driver.feature

```diff
@@ -0,0 +1,1265 @@
+#    When transaction rollbacks
+#    Then transaction is open: false
+#
+#    When connection open read transaction for database: typedb
+#    Then transaction has type: read
+#    Then transaction is open: true
+#    Then transaction rollbacks; fails
+#    Then transaction is open: false
+
+
+  # TODO: Check options setting and retrieval
+
+
+  # TODO: Fix on_close
+#  Scenario: Driver can schedule "transaction on close" jobs
+#    Given connection does not have database: created-after-schema
+#    Given connection does not have database: created-after-write
+#    Given connection does not have database: created-after-read
+#    When connection open schema transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-schema
+#    Then connection does not have database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open write transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-write
+#    Then connection has database: created-after-schema
+#    Then connection does not have database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#
+#    When connection open read transaction for database: typedb
+#    When schedule database creation on transaction close: created-after-read
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection does not have database: created-after-read
+#    When transaction closes
+#    Then connection has database: created-after-schema
+#    Then connection has database: created-after-write
+#    Then connection has database: created-after-read
+
+
+  Scenario: Driver processes transaction commit errors correctly
+    Given connection open schema transaction for database: typedb
+    When typeql schema query
+      """
+      define attribute name;
+      """
+    Then transaction commits; fails
```

Comment:
i feel like for these it would be worth having some kind of string error message to see if the driver got the right error message ( have been bitten by this before! )

Reply:
Sure, I'll add error messages now.

Reply:
Done! Added additional errors checking, remembered them in the morning.